### PR TITLE
Adding support for Android Q

### PR DESCRIPTION
**Pull request by bhariharan**
This PR:
- Bumps up `targetApiVersion` to `29`.
- Bumps up `minApiVersion` to `23`.
- Code refactoring to handle `TODO:` and `FIXME:` tags around conditional code to handle older versions prior to `23`.
- Removing methods that were marked `@deprecated` for `Mobile SDK 8.0` removal.

### CODE REVIEW

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java [hunk 1]
```diff
@@ -373,11 +374,6 @@ protected boolean onSubmit(String enteredPasscode) {
                 }
             }
             return true;
-
-        case Change:
```
Comment (brandonpage) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java]: Good catch.

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java [hunk 2]
```diff
@@ -261,18 +262,11 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
 		if (oldVersion == 1) {
 			SmartStore.createLongOperationsStatusTable(db);
 		}
-
```
Comment (wmathurin) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java]: If we are getting rid of this block, shouldn't we also get rid of the one above (line 261) ?

Comment (bhariharan) [reply libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java]: Yes - I could.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/security/KeyStoreWrapper.java [hunk 3]
```diff
@@ -227,33 +226,17 @@ private synchronized void createKeysIfNecessary(String algorithm, String name, i
...
-                         * Disabling StrongBox for now, since it's too slow on Pixel 3
-                         * and Pixel 3 XL.
-                         *
-                         * TODO: Re-enable in Mobile SDK 7.2.
-                         */
-                        //keyGenParameterSpecBuilder.setIsStrongBoxBacked(true);
-                        keyGenParameterSpecBuilder.setIsStrongBoxBacked(false);
-                        kpg.initialize(keyGenParameterSpecBuilder.build());
-                        kpg.generateKeyPair();
-                    } catch (StrongBoxUnavailableException sb) {
```
Comment (wmathurin) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/security/KeyStoreWrapper.java]: No need for the catch anymore?

Comment (bhariharan) [reply libs/SalesforceSDK/src/com/salesforce/androidsdk/security/KeyStoreWrapper.java]: Nope - we don't use `StrongBox` anymore based on Google's recommendation.

**Comment by bhariharan** [comment]
Looking into test failures.

**Comment by bhariharan** [comment]
All tests are passing now (was an unrelated token issue on `CircleCI`). Merging.

**Pull request merged**